src/Our.Umbraco.Gandalf/Core/Services/CachedProxies/AllowedIpServiceCachedProxy.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Collections.Generic;
+using Our.Umbraco.Gandalf.Core.Models.DTOs;
+using Our.Umbraco.OpenKeyValue.Core.Models.Pocos;
+using System.Linq;
+using Umbraco.Core.Cache;
+
+namespace Our.Umbraco.Gandalf.Core.Services.CachedProxies
+{
+	public class AllowedIpServiceCachedProxy : IAllowedIpService
+	{
+		private readonly IAllowedIpService _allowedIpService;
+		private readonly IAppCache _runtimeCache;
+
+		public AllowedIpServiceCachedProxy(AllowedIpService allowedIpService, AppCaches appCaches)
+		{
+			_allowedIpService = allowedIpService;
+			_runtimeCache = appCaches.RuntimeCache;
+		}
+	
+		public AllowedIpDto Create(string ipAddress, string notes)
+		{
+			_runtimeCache.ClearByRegex($"{typeof(AllowedIpServiceCachedProxy)}_(.*)");
+			return this._allowedIpService.Create(ipAddress, notes);
```

Comment:
You will need to swap these, as there can be concurrency issues when multiple backoffice users refresh the backoffice.

var result = this._allowedIpService.Create(ipAddress, notes);
_runtimeCache.ClearByRegex($"{typeof(AllowedIpServiceCachedProxy)}_(.*)");

return result;

Reply:
👍